Android/app/src/main/java/com/vigbag/android/SignInFragment.kt

```diff
@@ -0,0 +1,53 @@
+package com.vigbag.android
+
+import android.content.Context
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.google.android.material.snackbar.Snackbar
+import com.vigbag.android.databinding.FragmentSignInBinding
+
+class SignInFragment : Fragment() {
+    private lateinit var binding: FragmentSignInBinding
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        binding = FragmentSignInBinding.inflate(inflater, container, false)
+        validateCredentials()
+        return binding.root
+    }
+
+    private fun validateCredentials() {
+
+        val sharedPreferences = requireContext().getSharedPreferences("LoginPrefs", Context.MODE_PRIVATE)
```

Comment:
Try using Androidx.Preference library here. It has much simpler API. How?
1) Add this dependency in build.gradle: `implementation("androidx.preference:preference-ktx:1.2.0")`
2) Use `PreferenceManager.getDefaultSharedPreferences(requireContext())` in this line (22).